packages/design-system/src/components/web-components/ds-accordion/ds-accordion.test.tsx

```diff
@@ -1,35 +1,28 @@
-import { render } from '@testing-library/react';
+import { createTestRenderer } from '../__tests__/rendering';
 import './ds-accordion';
 import './ds-accordion-item';
 
-function renderAccordion(attrs = {}) {
-  return render(
-    <ds-accordion {...attrs}>
-      <ds-accordion-item heading="First amendment">
-        <p>We the People of the United States, in Order to form a more perfect Union...</p>
-      </ds-accordion-item>
-      <ds-accordion-item heading="Second amendment">
-        <p>A well regulated Militia, being necessary to the security of a free State...</p>
-      </ds-accordion-item>
-    </ds-accordion>
-  );
-}
+const renderAccordion = createTestRenderer('ds-accordion', (attrs = {}) => (
+  <ds-accordion {...attrs}>
+    <ds-accordion-item heading="First amendment">
+      <p>We the People of the United States, in Order to form a more perfect Union...</p>
+    </ds-accordion-item>
+    <ds-accordion-item heading="Second amendment">
+      <p>A well regulated Militia, being necessary to the security of a free State...</p>
+    </ds-accordion-item>
+  </ds-accordion>
+));
 
 describe('ds-accordion', () => {
-  it('renders a accordion item', () => {
-    const { asFragment } = renderAccordion();
-    expect(asFragment()).toMatchSnapshot();
-  });
-
   it('renders additional className', () => {
-    const { container } = renderAccordion({ 'class-name': 'ds-u-test' });
-    const accordion = container.querySelector('.ds-c-accordion');
+    const { shadowRoot } = renderAccordion({ 'class-name': 'ds-u-test' });
+    const accordion = shadowRoot.querySelector('.ds-c-accordion');
     expect(accordion.classList).toContain('ds-u-test');
   });
 
   it('renders ds-c-accordion--bordered class when a bordered prop is set', () => {
-    const { container } = renderAccordion({ bordered: 'true' });
-    const accordion = container.querySelector('.ds-c-accordion');
+    const { shadowRoot } = renderAccordion({ bordered: 'true' });
+    const accordion = shadowRoot.querySelector('.ds-c-accordion');
     expect(accordion.classList).toContain('ds-c-accordion--bordered');
```

Comment:
Do you think it would be worthwhile to also assert that an `accordion-item` child receives the correct `ds-c-accordion__content--bordered` style when the parent accordion has the bordered prop set?

Reply:
Oh good catch! I never added any unit tests related to that functionality

Reply:
Oh dang! Adding a unit test caught a bug. The `ds-accordion` component doesn't have a wrapper component that parses the `bordered` attribute into a boolean, so even values of `bordered="false"` give it a border!

Reply:
I think it's not happening in Storybook because it uses a boolean control in Storybook and ends up providing the component with a real boolean value in the web component that is rendered by Preact (which is not an expected real-world scenario)